server/app/repository/ProgramRepository.java

```diff
@@ -173,6 +173,32 @@ public CompletableFuture<Program> getActiveProgramFromSlug(String slug) {
         executionContext.current());
   }
 
+  /** Get the current draft program with the provided slug. */
+  public CompletableFuture<Program> getDraftProgramFromSlug(String slug) {
+    return supplyAsync(
+        () -> {
+          for (Program program : database.find(Program.class).where().isNull("slug").findList()) {
+            program.getSlug();
+            program.save();
+          }
```

Comment:
**issue (blocking):** This is unrelated to the name and javadoc of the method. In general reads should not alter database state unless they're explicitly named `findOrCreate`. If it's necessary for the feature to work please refactor for clarity, otherwise remove.

Reply:
🙈 that is not going to make it easier for me to sleep this weekend. Thanks for pointing that out.

Reply:
Looking at Program and ProgramDefinition, I think this was added originally because the `slug` column wasn't populated for all programs at one point, so this code was included as a hacky way of doing a data migration. All new programs have the column populated though and it's been a while, so I think it's safe to remove this code.

Reply:
I'll do this in a separate PR so that it's easy to roll back if necessary.

Reply:
https://github.com/civiform/civiform/pull/5321